src/Router.js

```diff
@@ -160,6 +161,10 @@ export default class Router {
 
       this.__currentCanonicalPath = canonicalPath
 
+      if (this.onRouteStart && mode !== 'replace') {
+        this.onRouteStart();
```

Comment:
Yeah if we have Route A redirecting to Route B:

- Route A start (should call onStart)
- Route A redirect
- Route A end (never called)
- Route B start (should not call onStart)
- Route B end (should call onEnd)